llvm/lib/Transforms/IPO/WholeProgramDevirt.cpp

```diff
@@ -1456,6 +1456,12 @@ void DevirtModule::tryICallBranchFunnel(
   if (!HasNonDevirt)
     return;
 
+  // If any GV is AvailableExternally, drop to generate branch.funnel
```

Comment:
Change "drop to generate" to "do not generate". Also, please add a note that this avoids a crash in LowerTypeTest due to the GV being dropped resulting in a null GlobalTypeMember.